things/mocks/auth.go

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package mocks
+
+import (
+	"context"
+
+	"github.com/mainflux/mainflux"
+	"github.com/mainflux/mainflux/pkg/errors"
+	"github.com/stretchr/testify/mock"
+	"google.golang.org/grpc"
+)
+
+var _ mainflux.AuthzServiceClient = (*Service)(nil)
+
+type Service struct {
+	mock.Mock
+}
+
+func (m *Service) Authorize(ctx context.Context, in *mainflux.AuthorizeReq, opts ...grpc.CallOption) (*mainflux.AuthorizeRes, error) {
+	ret := m.Called(ctx, in)
+	if in.GetSubject() == WrongID || in.GetSubject() == "" {
+		return &mainflux.AuthorizeRes{}, errors.ErrAuthorization
+	}
+	if in.GetObject() == WrongID || in.GetObject() == "" {
+		return &mainflux.AuthorizeRes{}, errors.ErrAuthorization
+	}
+
```

Comment:
this could be defined in On() rather then modifying the mocks

Reply:
This is done at https://github.com/mainflux/mainflux/pull/1927. The reason I implemented it is to make code `DRY`. Imagine doing it for all test cases, yet it can be done at one place.

Reply:
It is OK to have it like this since we have a deterministic auth mock and we do not have to repeat it in cases, just like @rodneyosodo mentioned.